editoast/src/views/study.rs

```diff
@@ -472,85 +441,172 @@ pub mod test {
+            "test_study_name",
+            created_project.id,
+        )
+        .await;
 
-        let req = TestRequest::get().uri(url.as_str()).to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::OK);
+        let request = TestRequest::delete()
+            .uri(&format!(
+                "/projects/{}/studies/{}",
+                created_project.id, created_study.id
+            ))
+            .to_request();
 
-        let conn = &mut db_pool.get().await.unwrap();
-        assert!(Study::delete_static(conn, study_fixture_set.study.id())
+        let response = call_service(&app.service, request).await;
+        assert_eq!(response.status(), StatusCode::NO_CONTENT);
+
+        let exists = Study::exists(db_pool.get_ok().deref_mut(), created_study.id)
             .await
-            .unwrap());
+            .expect("Failed to check if study exists");
 
-        let req = TestRequest::get().uri(url.as_str()).to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::NOT_FOUND);
+        assert!(!exists);
     }
 
     #[rstest]
-    async fn study_patch(#[future] study_fixture_set: StudyFixtureSet) {
-        let app = create_test_service().await;
-        let study_fixture_set = study_fixture_set.await;
-        let req = TestRequest::patch()
-            .uri(easy_study_url(&study_fixture_set, true).as_str())
-            .set_json(json!({"name": "rename_test", "budget":20000}))
-            .to_request();
-        let response = call_service(&app, req).await;
-        assert_eq!(response.status(), StatusCode::OK);
+    async fn study_patch() {
+        let app = TestAppBuilder::default_app();
+        let db_pool = app.db_pool();
+
+        let created_project =
+            create_project(db_pool.get_ok().deref_mut(), "test_project_name").await;
+
+        let created_study = create_study(
+            db_pool.get_ok().deref_mut(),
+            "test_study_name",
+            created_project.id,
+        )
+        .await;
 
-        let StudyWithScenarioCount { study, .. } = read_body_json(response).await;
-        assert_eq!(study.name, "rename_test");
+        let study_name = "rename_test";
+        let study_budget = 20000;
+
+        let request = TestRequest::patch()
+            .uri(&format!(
+                "/projects/{}/studies/{}",
+                created_project.id, created_study.id
+            ))
+            .set_json(json!({
+                "name": study_name,
+                "budget": study_budget,
+            }))
+            .to_request();
+        let response: StudyResponse = call_and_read_body_json(&app.service, request).await;
+
+        assert_eq!(response.study.name, study_name);
+        assert_eq!(response.study.budget, Some(study_budget));
+        assert_eq!(response.study.project_id, created_project.id);
+        assert_eq!(response.project.id, created_project.id);
```

Comment:
Probably better `retrieve` and check that both the project and the study are equal to what's been returned by the response. 
(We could build the response within a transaction where all db operations have been performed, forget to commit it and return the response anyway. All these checks would hold but the study wouldn't have been patched.)